src/node_crypto.cc

```diff
@@ -1537,7 +1537,12 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
       (void) BIO_reset(bio);
 
       BN_ULONG exponent_word = BN_get_word(rsa->e);
-      BIO_printf(bio, "0x%lx", exponent_word);
+# ifdef SIXTY_FOUR_BIT_LONG
+        BIO_printf(bio, "0x%lx", exponent_word);
+# endif
+# ifdef THIRTY_TWO_BIT
+        BIO_printf(bio, "0x%dx", exponent_word);
+# endif
```

Comment:
A less ifdef-y approach would be BIO_printf(bio, "0x%llx", static_cast<uint64_t>(exponent_word));